dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1274 @@
+
+* 规则描述：
+
+    没有合适的 store 用来补副本。
+
+* 处理方法：
+
+    * 检查 store 是否空间不足。
+    * 根据 label 配置（如果有这个配置的话）来检查是否有可以补副本的 store。
+
+## TiKV 报警规则
+
+本节介绍了 TiKV 组件的报警项。根据严重级别，报警项可分为三类，按照严重程度由高到低依次为：紧急级别、重要级别、警告级别。
+
+### 紧急级别报警项
+
+紧急级别的报警通常由于服务停止或节点故障导致，此时需要马上进行人工干预操作。
+
+#### `TiKV_memory_used_too_fast`
+
+* 报警规则：
+
+    `process_resident_memory_bytes{job=~"tikv",instance=~".*"} - (process_resident_memory_bytes{job=~"tikv",instance=~".*"} offset 5m) > 5*1024*1024*1024`
+
+* 规则描述：
+
+    目前没有和内存相关的 TiKV 的监控，你可以通过 node_exporter 监控集群内机器的内存使用情况。如上规则表示，如果在 5 分钟之内内存使用超过 5GB（TiKV 内存占用的太快），则报警。
+
+* 处理方法：
+
+    调整 `rockdb.defaultcf` 和 `rocksdb.writecf` 的 `block-cache-size` 的大小。
+
+#### `TiKV_GC_can_not_work`
+
+* 报警规则：
+
+    `sum(increase(tidb_tikvclient_gc_action_result{type="success"}[6h])) < 1`
+
+* 规则描述：
+
+    在 6 小时内 Region 上没有成功执行 GC，说明 GC 不能正常工作了。短期内 GC 不运行不会造成太大的影响，但如果 GC 一直不运行，版本会越来越多，从而导致查询变慢。
+
+* 处理方法：
+
+    1. 执行 `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME="tikv_gc_leader_desc"` 来找到 gc leader 对应的 `tidb-server`；
+    2. 查看该 `tidb-server` 的日志，grep gc_worker tidb.log；
+    3. 如果发现这段时间一直在 resolve locks（最后一条日志是“start resolve locks”）或者 delete ranges（最后一条日志是“start delete {number} ranges”），说明 GC 进程是正常的。否则需要联系 `support@pingcap.com` 帮忙处理。
+
+### 重要级别报警项
+
+对于重要级别的报警，需要密切关注异常指标。
+
+#### `TiKV_server_report_failure_msg_total`
+
+* 报警规则：
+
+    `sum(rate(tikv_server_report_failure_msg_total{type="unreachable"}[10m])) BY (store_id) > 10`
+
+* 规则描述：
+
+    表明无法连接远端的 TiKV。
+
+* 处理方法：
+
+    1. 检查网络是否通畅。
+    2. 检查远端 TiKV 是否挂掉。
+    3. 如果远端 TiKV 没有挂掉，检查压力是否太大，参考 [`TiKV_channel_full_total`](#-tikv-channel-full-total) 处理方法。
+
+#### `TiKV_channel_full_total`
+
+* 报警规则：
+
+    `sum(rate(tikv_channel_full_total[10m])) BY (type, instance) > 0`
+
+* 规则描述：
+
+    该错误通常是因为 Raftstore 线程卡死，TiKV 的压力已经非常大了。
+
+* 处理方法：
+
+    1. 观察 Raft Propose 监控，看这个报警的 TiKV 节点是否明显有比其他 TiKV 高很多。如果是，表明这个 TiKV 上有热点，需要检查热点调度是否能正常工作。
+    2. 观察 Raft IO 监控，看延迟是否升高。如果延迟很高，表明磁盘可能有瓶颈。一个能缓解但不怎么安全的办法是将 `sync-log` 改成 `false`。
+    3. 观察 Raft Process 监控，看 tick duration 是否很高。如果是，需要在 `[raftstore]` 配置下加上 `raft-base-tick-interval = “2s”`。
+
+#### `TiKV_write_stall`
+
+* 报警规则：
+
+    `delta(tikv_engine_write_stall[10m]) > 0`
+
+* 规则描述：
+
+    RocksDB 写入压力太大，出现了 stall。
+
+* 处理方法：
+
+    1. 观察磁盘监控，排除磁盘问题。
+    2. 看 TiKV 是否有写入热点。
+    3. 在 `[rocksdb]` 和 `[raftdb]` 配置下调大 `max-sub-compactions` 的值。
+
+#### `TiKV_raft_log_lag`
+
+* 报警规则：
+
+    `histogram_quantile(0.99, sum(rate(tikv_raftstore_log_lag_bucket[1m])) by (le, instance)) > 5000`
+
+* 规则描述：
+
+    这个值偏大，表明 Follower 已经远远落后于 Leader，Raft 没法正常同步了。可能的原因是 Follower 所在的 TiKV 卡住或者挂掉了。
+
+#### `TiKV_async_request_snapshot_duration_seconds`
+
+* 报警规则：
+
+    `histogram_quantile(0.99, sum(rate(tikv_storage_engine_async_request_duration_seconds_bucket{type="snapshot"}[1m])) by (le, instance, type)) > 1`
+
+* 规则描述：
+
+    这个值偏大，表明 Raftstore 负载压力很大，可能已经卡住。
+
+* 处理方法：
+
+    参考 [`TiKV_channel_full_total`](#-tikv-channel-full-total) 的处理方法。
```

Comment:
```suggestion
    参考 [`TiKV_channel_full_total`](#tikv-channel-full-total) 的处理方法。
```

Reply:
Ref: you can test the link rule [here](https://pingcap.com/docs/v3.0/reference/configuration/tidb-server/configuration/#config) (an example).